packages/ui5-config/src/middlewares.ts

```diff
@@ -128,23 +129,30 @@ export function getFioriToolsProxyMiddlewareConfig(
 }
 
 export const getMockServerMiddlewareConfig = (
-    path?: string,
-    annotationsConfig: MockserverConfig['annotations'] = []
+    dataSourcesConfig: DataSourceConfig[],
+    annotationsConfig: MockserverConfig['annotations'],
+    appRoot?: string
 ): CustomMiddleware<MockserverConfig> => {
-    path = path?.replace(/\/$/, ''); // Mockserver is sensitive to trailing '/'
+    const services: MockserverConfig['services'] = [];
+
+    // Populate services based on dataSourcesConfig
+    dataSourcesConfig.forEach((dataSource) => {
+        const serviceRoot = `${appRoot ?? './webapp'}/localService/${dataSource.serviceName}`;
```

Comment:
Can't we leave this as it is for now ?
If I use that function from project-acces, Typescript throws me cycle error: **Project references may not form a circular graph. Cycle detected.**
`project-access` uses `ui5-config` and `ui5-config` uses `project-access`.

![image](https://github.com/user-attachments/assets/0d3df27e-d137-40a2-ae21-3b0a2ea813da)

![image](https://github.com/user-attachments/assets/a4d5d3cd-d984-4a4b-9709-f1c180867253)

Reply:
Yes. Let's ignore this for now because that will hopefully be solved with https://github.com/SAP/open-ux-tools/issues/2531 because ui5-config actually should not have dependencies to middlewares (only the other way around).